draft-ietf-tls-ticketrequests.md

```diff
@@ -180,38 +179,56 @@ ultimately end up with just the tickets from a single resumed session, so
 in that case a resumption_count commensurate with the number of parallel
 sessions would be used.
 
+Priming of the ticket cache need not happen all at once.  Clients that
+anticipate opening frequent connections to a given server can ask for 2 or 3
+tickets on each full handshake, and attain the requisite steady-state cache
+size after a few initial full handshakes, without substantial risk of
+"overshoot" or need to accurately predict the connection concurrency.
+
+When a client presenting a previously obtained ticket finds that the server
+nevertheless negotiates a fresh session, the client should assume that any
+other tickets associated with the same session as the presented ticket are also
+no longer valid for resumption.  This includes not only the tickets obtained
+during the initial full handshake, but all tickets subsequently obtained as
+part of subsequent resumptions.  Requesting more than one ticket in case a
+full handshake is forced by the server helps to keep the session cache primed.
+
 On the other hand, with e.g. server-to-server traffic with fixed source
-addresses and no connection racing, ticket reuse may be appropriate.  Clients
-that wish to reuse tickets when possible should send an extension with the
-resumption_count set to 0, and the new_session_count set to the value desired
-for a full handshake (typically 1).
+addresses and no connection racing, ticket reuse may be an option (ticket reuse
+is NOT RECOMMENDED per Section C.4 of {{RFC8446}}, and is only appropriate when
+replay and privacy concerns do not apply).  Clients that wish to reuse tickets
+when possible should send an extension with the resumption_count set to 0, and
+the new_session_count set to the value desired for a full handshake (typically
+1).
 
 Servers SHOULD NOT send more tickets than requested for the handshake type
 selected by the server (resumption or full handshake) as clients will most
 likely discard any additional tickets.
 
-When a resumption is performed, and the requested resumption_count is zero, the
-client is requesting ticket reuse.  If the presented ticket is valid for
-additional resumptions (only possible when the server supports reuse), it
-SHOULD return no tickets.  Otherwise, when either the server does not support
-reuse, or the ticket is expiring, and needs to be replaced (e.g. reissued under
-a fresh session-ticket encryption key) the server SHOULD return one ticket.
+When the client presents a ticket for resumption, but the requested
+resumption_count is zero and the new_session_count is non-zero, the client
+is requesting ticket reuse (despite section C.4 of {{RFC8446}}).  If
+the presented ticket is valid for additional resumptions (only possible when
+the server supports reuse), it SHOULD return no tickets.  Otherwise, when
+either the server does not support reuse, or the ticket is expiring, and needs
+to be replaced (e.g. reissued under a fresh session-ticket encryption key) the
+server SHOULD return one ticket.
```

Comment:
I think (on first glance) I'm personally okay with this logic, but we'll need input from the WG, of course.